src/manageiq_client/api.py

```diff
@@ -457,11 +482,18 @@ def __init__(self, obj):
 
     def reload(self):
         self._obj.reload_if_needed()
+        reloaded_actions = []
         for action in self._obj._actions:
-            setattr(
-                self,
-                action["name"],
-                Action(self, action["name"], action["method"], action["href"]))
+            # There can be multiple actions with the same name and different methods
+            # (e.g. actions "delete" with method POST and DELETE).
+            # This makes sure that the attribute refers to the first action and is not redefined
+            # by other action with the same name and different method.
```

Comment:
The thing is that this situation applies just to action "delete" with method POST and action "delete" with method DELETE. Sometimes there are both, sometimes just the one with POST (usually there are both for "entities" and just "delete" with POST for collections).
This makes sure that we are always testing the same thing (i.e. the one with POST) and that we have the possibility to override it if we also want to test the DELETE method.
At the moment there is no other "clash" like this one, so I thought that such simple scenario doesn't justify more complex and more "generic" approach.